moped-editor/src/views/projects/projectView/ProjectComponents/BasemapSpeedDial.js

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from "react";
+import clsx from "clsx";
+import { Icon, makeStyles, Typography } from "@material-ui/core";
+import { SpeedDial, SpeedDialAction } from "@material-ui/lab";
+import {
+  MAPBOX_PADDING_PIXELS,
+  MAPBOX_CONTROL_BUTTON_WIDTH,
+  COLORS,
+} from "./mapStyleSettings";
+
+const useStyles = makeStyles((theme) => ({
+  speedDialAction: {
+    width: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    height: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    backgroundSize: "100% 100%",
+    borderRadius: 4,
+    background: COLORS.mutedGray,
+  },
+  speedDialStreets: {
+    color: "black",
+    backgroundImage: `url(${process.env.PUBLIC_URL}/static/images/mapStreets.jpg)`,
+  },
+  speedDialAerial: {
+    color: "white",
+    backgroundImage: `url(${process.env.PUBLIC_URL}/static/images/mapAerial.jpg)`,
+  },
+  fabLabel: ({ isSpeedDialOpen }) => ({
+    ...(isSpeedDialOpen && { backgroundColor: "rgba(0,0,0,.25)" }),
+    height: "100%",
+  }),
+  mapStyleToggleLabel: {
+    fontSize: ".8rem",
+    fontWeight: "bold",
+  },
+  mapStyleToggleLabelIcon: {
+    position: "relative",
+    top: 3,
+    marginRight: "4px",
+  },
+  mapStyleActionLabel: {
+    position: "absolute",
+    bottom: 0,
+  },
+  speedDial: {
+    position: "absolute",
+    height: MAPBOX_CONTROL_BUTTON_WIDTH * 2,
+    right: `${MAPBOX_PADDING_PIXELS}px`,
+    // Mapbox basemap has copyright info below the speedial while NearMap tiles do not
+    bottom: ({ basemapKey }) =>
+      basemapKey === "aerial"
+        ? `${MAPBOX_PADDING_PIXELS}px`
+        : `${MAPBOX_PADDING_PIXELS * 3}px`,
+    // Mapbox copyright info collapses to a taller info icon at 990px and below
+    [theme.breakpoints.down(991)]: {
+      bottom: ({ basemapKey }) =>
+        basemapKey === "aerial"
+          ? `${MAPBOX_PADDING_PIXELS}px`
+          : `${MAPBOX_PADDING_PIXELS * 4}px`,
+    },
+  },
+}));
+
+/**
+ * Generates the basemap selector using the SpeedDial component
+ * @param {function} setBasemapKey - The function we will call with to change the basemap key value
+ * @param {string} basemapKey - The name of the current basemap key
+ * @return {JSX.Element}
+ */
+const BasemapSpeedDial = ({ setBasemapKey, basemapKey }) => {
+  const [isSpeedDialOpen, setIsSpeedDialOpen] = useState(false);
+  const classes = useStyles({ basemapKey, isSpeedDialOpen });
+
+  /**
+   * Changes the current basemap and closes the speed dial menu
+   * @param basemapName
+   */
+  const onBasemapSelect = (basemapKey) => {
+    setIsSpeedDialOpen(false);
+    setBasemapKey(basemapKey);
+  };
+
+  /**
+   * Opens the speed dial menu
+   */
+  const onOpen = () => {
+    setIsSpeedDialOpen(true);
+  };
+
+  const onClose = () => setIsSpeedDialOpen(false);
+
+  return (
+    <SpeedDial
```

Comment:
The speed dial opens `onMouseover`, and it opens/closes onClick. For those whose instinct is to click on it, you end up opening on mouseover and then immediately closing it onClick. What would you think about disabling the click handler entirely, provided you get a little grace to mouse from the speed dial to the basemap square to make your choice.

Reply:
I agree with this but it is the default Material UI behavior so I'll stick with it unless we get some user feedback about it. TIL because I always thought more than the appearance was customized when this component was originally written. https://mui.com/material-ui/react-speed-dial/